examples/graphql-interceptor-configurations/graphql_interceptor_configurations.bal

```diff
@@ -0,0 +1,56 @@
+import ballerina/graphql;
+import ballerina/log;
+import ballerina/time;
+
+// Configure the behavior of the interceptor.
+@graphql:InterceptorConfig {
+    // Change the scope of the interceptor.
+    global: false
+}
+readonly service class LogExecutionTime {
+    *graphql:Interceptor;
+
+    isolated remote function execute(graphql:Context context, graphql:Field 'field)
+    returns anydata|error {
+        // Access the current execution field name using the `graphql:Field` object.
+        string fieldName = 'field.getName();
+
+        // Get execution start time.
+        time:Utc startUtc = time:utcNow();
+
+        // Invoke the next interceptor or resolver.
+        var data = context.resolve('field);
+
+        // Get execution end time.
+        time:Utc endUtc = time:utcNow();
+
+        // Calculate the execution time.
+        time:Seconds executionTime = time:utcDiffSeconds(endUtc, startUtc);
+
+        // Log the execution time.
+        log:printInfo(string `execution time of the "${fieldName}" field: ${executionTime}s`);
+
+        // Returns the execution result of the next interceptor or the resolver.
+        return data;
+    }
+}
+
+// Define the type Name.
+public type Name record {|
+    string first;
+    string last;
+|};
```

Comment:
A simple greeting service would be enough for this scenario, right?

Reply:
A type with two subfields was added to specify that the service interceptor will not execute for subfields when `global=false`.